sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala

```diff
@@ -63,7 +63,10 @@ case class CreateFunctionCommand(
     } else {
       // For a permanent, we will store the metadata into underlying external catalog.
       // This function will be loaded into the FunctionRegistry when a query uses it.
-      // We do not load it into FunctionRegistry right now.
+      // We do not load it into FunctionRegistry right now. However, we need to validate.
+      catalog.loadFunctionResources(resources)
+      catalog.makeFunctionBuilder(functionName, className)
```

Comment:
Actually, this code is the same with [the previous lines 59~61](https://github.com/dongjoon-hyun/spark/blob/7641789c00360ed1ba0fe4475b350415fea56c95/sql/core/src/main/scala/org/apache/spark/sql/execution/command/functions.scala#L59-L61).

Putting aside this two lines, I think we can reuse at least the fixes on test cases for refactoring. Are you planning the refactoring before Spark 2.2.0 branch cut?